adapters/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -86,6 +89,7 @@ private void removeListener(OrderedRealmCollection<T> data) {
      * @return the number of items.
      */
     @Override
+    @CheckResult
```

Comment:
> I am assuming because actually checking the constraint isn't free either.

What exactly do you mean with this? The annotations are only evaluated by IntelliJ/AndroidStudio and at compile time. At runtime they aren't present.

I agree with you, that annotating `get*`-methods with `@CheckResult` might be a bit to much. Should I remove those annotations?